### PR TITLE
Ensure Dead() doesn't block when no goroutines are spawned

### PR DESCRIPTION
**Pull request by vincentbernat**
This is done by closing the dead channel and creating a new one on the
first spawned goroutine. There are several problems with this
approach:

 1. Tests are expecting `Dead()` to block. I didn't "correct" the
    tests as it seems to be a strong assumption.

 2. People may expect `Dead()` to block even without any
    goroutine. One may start an unmanaged goroutine using `Dead()` and
    then only spawn goroutines with the tomb. Before this change, no
    problem. After this change, the unmanaged goroutine will not be
    blocked by `Dead()`.

Related to #17.

So, in conclusion, there is no good way around that. Maybe just document that at least a goroutine is needed to get the expected behavior. I am making this PR as a documentation for my tentative. Feel free to close it.

### CODE REVIEW

**Comment by niemeyer** [comment]
This code means the dead channel is _always_ closed. That makes it not very useful.

Please see comments in #17 as well.

**Comment by vincentbernat** [comment]
Just a nitpick, but the dead channel is closed after init() and "open" just before the first goroutine is spawned.

**Comment by niemeyer** [comment]
That's a race condition.. for anything reaching the channel before the first goroutine is spawned, it will be always closed, and as long as you have concurrent code running, we can't tell who picked up the first channel or the second channel.

**Comment by niemeyer** [comment]
For those watching, please see #17 for more details. As a rule of thumb, if you put a tomb in use, make sure the work is eventually started. The whole point of the package is to track the termination of that work, and it will protect against race conditions for that specifically. If you don't plan on starting the work, just don't share the tomb around.
